channels/ON-CHAIN.md

```diff
@@ -102,13 +103,18 @@ This operation is not mandatory for normal channel operation.
 Serialization defined [here](../serializations.md#channel-deposit-transaction)
 
 
-- `channel_id`: channel id as recorded on chain
+- `channel_id`: channel id as recorded on-chain
 - `from`: sender of the deposit
 - `amount`: amount of tokens deposited
 - `ttl`:
 - `fee`:
+- `state_hash`: the root hash of the channel state tree after the deposit had been applied
```

Comment:
This is a transaction for depositing funds. How can this hash be a root hash *after* deposit applied?

If meant to be how root hash would be after application of tx if tx were accepted, which state shall be considered to be the starting point?

Reply:
It is the internal channel's state tree being applied the deposit transaction.
This hash is not validated on-chain, just persisted. It could be any valid hash as a snapshot.
If it were to be used later for closing the channel, the poster party must provide a corresponding PoI with the exact same root hash.

Reply:
> internal channel's state tree

Ah ok. So it is not a [*the* channel state tree](https://github.com/aeternity/protocol/blame/0f33f73768bba3931c0835b43fcf08800bbe0422/serializations.md#L523) but an internal one, a new entity. And I see some definition [here](https://github.com/aeternity/protocol/pull/119/files#diff-4fbc1ee2b0f1c1c88eeeafc062a7989cR202). Ok.